crates/cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -120,47 +132,35 @@ impl ParsedQObject {
         })
     }
 
-    fn parse_qml_metadata(
-        name: &Name,
-        attrs: &mut Vec<Attribute>,
-    ) -> Result<Option<QmlElementMetadata>> {
-        // Find if there is a qml_element attribute
-        if let Some(attr) = attribute_take_path(attrs, &["qml_element"]) {
+    fn parse_qml_metadata(name: &Name, attrs: &[Attribute]) -> Result<Option<QmlElementMetadata>> {
+        let attributes = parse_attributes(attrs, &Self::ALLOWED_ATTRS)?;
+        if let Some(attr) = attributes.get("qml_element") {
             // Extract the name of the qml_element from macro, else use the c++ name
             // This will use the name provided by cxx_name if that attr was present
-            let name = match attr.meta {
+            let name = match &attr.meta {
                 Meta::NameValue(name_value) => expr_to_string(&name_value.value)?,
                 _ => name.cxx_unqualified(),
             };
-
-            // Determine if this element is uncreatable
-            let uncreatable = attribute_take_path(attrs, &["qml_uncreatable"]).is_some();
-
-            // Determine if this element is a singleton
-            let singleton = attribute_take_path(attrs, &["qml_singleton"]).is_some();
-
+            let uncreatable = attributes.contains_key("qml_uncreatable");
+            let singleton = attributes.contains_key("qml_singleton");
             return Ok(Some(QmlElementMetadata {
                 name,
                 uncreatable,
                 singleton,
             }));
         }
-
         Ok(None)
     }
 
     fn parse_property_attributes(attrs: &mut Vec<Attribute>) -> Result<Vec<ParsedQProperty>> {
-        let mut properties = vec![];
-
-        // Note that once extract_if is stable, this would allow for comparing all the
-        // elements once using path_compare_str and then building ParsedQProperty
-        // from the extracted elements.
+        // Once extract_if is stable, this would allow comparing all the elements using
+        // path_compare_str and building ParsedQProperty from the extracted elements.
         // https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.extract_if
-        while let Some(attr) = attribute_take_path(attrs, &["qproperty"]) {
-            properties.push(ParsedQProperty::parse(attr)?);
-        }
-
-        Ok(properties)
+        attrs
+            .drain(..)
```

Comment:
wonder if this should be `iter()` as this modifies the attribute list rather than parsing and rebuilding which is the new philosophy?

Reply:
ParsedQProperty::parse needs to take an owned attr, so using iter creates a reference, and it cannot be moved. So either a mutable reference and drain needs to be used, or attr needs to be cloned I think

Reply:
Probably clone it into the ParsedQProperty::parse then, or change ParsedQProperty to take a reference (but what do other parse's do?)